diff --git a/Cargo.lock b/Cargo.lock
index 865e3d1c2..1684ca141 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -7122,7 +7122,6 @@ dependencies = [
  "sp1-prover",
  "sp1-sdk",
  "sp1-stark",
- "test-artifacts",
  "time 0.3.41",
  "tracing",
 ]
diff --git a/crates/perf/Cargo.toml b/crates/perf/Cargo.toml
index 703c10c39..1195f1511 100644
--- a/crates/perf/Cargo.toml
+++ b/crates/perf/Cargo.toml
@@ -19,7 +19,7 @@ sp1-sdk = { workspace = true }
 p3-baby-bear = { workspace = true }
 sp1-stark = { workspace = true }
 sp1-cuda = { workspace = true }
-test-artifacts = { workspace = true }
+# test-artifacts = { workspace = true }
 tracing = { workspace = true }
 serde_json = { workspace = true }
 rand = "0.8.5"
diff --git a/crates/perf/src/executor.rs b/crates/perf/src/executor.rs
index c91c050d9..a3ed5bdd2 100644
--- a/crates/perf/src/executor.rs
+++ b/crates/perf/src/executor.rs
@@ -1,6 +1,6 @@
 use std::time::{Duration, Instant};
 
-use clap::{command, Parser};
+use clap::Parser;
 use p3_baby_bear::BabyBear;
 use sp1_core_executor::{Executor, ExecutorMode, Program};
 use sp1_core_machine::shape::CoreShapeConfig;
diff --git a/crates/perf/src/main.rs b/crates/perf/src/main.rs
index 1944db771..4f8e4e221 100644
--- a/crates/perf/src/main.rs
+++ b/crates/perf/src/main.rs
@@ -6,7 +6,7 @@ use sp1_cuda::{MoongateServer, SP1CudaProver};
 use sp1_prover::{components::CpuProverComponents, HashableKey, ProverMode};
 use sp1_sdk::{self, Prover, ProverClient, SP1Context, SP1Prover, SP1Stdin};
 use sp1_stark::SP1ProverOpts;
-use test_artifacts::VERIFY_PROOF_ELF;
+// use test_artifacts::VERIFY_PROOF_ELF;
 
 #[derive(Parser, Clone)]
 #[command(about = "Evaluate the performance of SP1 on programs.")]
